src/Stack.php

```diff
@@ -16,9 +17,9 @@ class Stack
     /**
      * @param array $middleware
      */
-    public function __construct(array $middleware = [])
+    public function __construct(...$middleware)
     {
-        array_map([$this, 'append'], $middleware);
```

Comment:
You lost this, which destroys the ability to ensure type safety of middleware added by the constructor.